### PR TITLE
make process-compose-flake usable without flake-parts

### PR DESCRIPTION
**Pull request by VanCoding**
See previous discussion in #79 

followup PRs are in branches [VanCoding:step2](https://github.com/VanCoding/process-compose-flake/tree/step2) and [VanCoding:step3](https://github.com/VanCoding/process-compose-flake/tree/step3)

### CODE REVIEW

File: flake.nix
Comment (srid) [inline flake.nix:6]: Since `lib` already exists for this purpose, you want to put those two functions inside of it.

File: nix/eval-modules.nix
Comment (srid) [inline nix/eval-modules.nix:2]: Could you add document the function as a comment above? Here, and in `makeProcessCompose` function below.

**Review by srid — CHANGES_REQUESTED** [review]
Looks good; just a couple of changes.

**Comment by srid** [comment]
Also, CI is failing.

File: example/flake.nix
Comment (srid) [inline example/flake.nix:81]: This would look something like:

```suggestion
        packages.ponysay = inputs.process-compose-flake.lib.makeProcessCompose {
```

File: flake.nix
Comment (VanCoding) [reply flake.nix:6]: What bugs me a bit about this is that if we put the functions inside lib, we first need to pass in "lib", and then pass in "pkgs" again which already has lib. This makes using the function a bit strange, but I can do it if you want.

File: example/flake.nix
Comment (VanCoding) [reply example/flake.nix:81]: Doesn't this mean that the example can only be run *after* pushing the code upstream? I think it's nice if it's possible to use the example while working on the code. But I can surely do it.

Comment (srid) [reply example/flake.nix:81]: I'll update the lock file after merging the PR, but in CI, it will always use latest, due to https://github.com/Platonic-Systems/process-compose-flake/blob/f6ce9481df9aec739e4e06b67492401a5bb4f0b1/flake.nix#L15

File: flake.nix
Comment (srid) [reply flake.nix:6]: You make the arguments like this:

```nix
{ pkgs, lib ? pkgs.lib, ... }
```

Then only one argument (the first) is non-optional.

File: example/flake.nix
Comment (srid) [reply example/flake.nix:81]: In the meanwhile, you can do `cd ./example && nix run . --override-input process-compose-flake ..` to test locally.

File: flake.nix
Comment (VanCoding) [reply flake.nix]: Ok, done. This is a breaking change, though.

File: nix/eval-modules.nix
Comment (VanCoding) [reply nix/eval-modules.nix]: Done.

File: example/flake.nix
Comment (VanCoding) [reply example/flake.nix]: Done.

**Review by srid — COMMENTED** [review]
LGTM. Will merge once CI failures are addressed.

**Comment by VanCoding** [comment]
> In the meanwhile, you can do cd ./example && nix run . --override-input process-compose-flake .. to test locally.

This didn't work for me, it still uses lib from upstream. But I assume it'll work now.

**Comment by VanCoding** [comment]
@srid Nice!! Thanks for pulling that one in. :)

**Pull request merged**
